internal/emlb/emlb.go

```diff
@@ -247,18 +247,18 @@ func (e *EMLB) DeleteLoadBalancer(ctx context.Context, clusterScope *scope.Clust
 		return nil
 	}
 
-	log.Info("Deleting EMLB", "Cluster Metro", e.metro, "Cluster Name", clusterName, "Project ID", e.projectID, "Load Balancer ID", lbID)
-
 	// Fetch the Load Balancer object.
 	// Skip if 404, otherwise error
-	lb, err := e.getLoadBalancer(ctx, lbID)
-	if err != nil {
-		log.Error(err, "failed to load the loadbalancer object, cannot proceed with deletion")
+	lb, resp, err := e.getLoadBalancer(ctx, lbID)
+	if err != nil && (resp.StatusCode != http.StatusNotFound) {
```

Comment:
We could `return nil` early when the 404 is detected rather than attempt to delete the resource that we are pretty sure does not exist.  That said, the delete handling is also looking for a 404, so assuming that is what EMLB returns in this scenario, it should be fine.

```suggestion
	if err != nil {
		if (resp.StatusCode == http.StatusNotFound) {
			return nil
		} else {
```

Reply:
Err. We do need to exit. I didn't notice when commenting about that the `deleteLoadBalancer` is performed against `lb.GetId()` (the result of the failed `GET`) vs `lbID`.

If you want to take the existing approach, you'll need to avoid depending on `lbID`.

Why are we calling `getLoadBalancer` at all?